File: 백준/Silver II/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringBuilder sb = new StringBuilder();
	static StringTokenizer st;
	static int n,m;
	static List[] g;
	static boolean[] visit;
	public static void main(String[] args) throws IOException {
		st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken())+1;
		m = Integer.parseInt(st.nextToken());
		int v = Integer.parseInt(st.nextToken());
		g = new List[n];
		for (int j=0; j<n; j++)
			g[j] = new ArrayList<>();
		for (int i=0; i<m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			g[a].add(b);
			g[b].add(a);
		}
		for (int j=0; j<n; j++)
			Collections.sort(g[j]);
		visit = new boolean[n];
		dfs(0,v);
		sb.append("\n");
		visit = new boolean[n];
		bfs(v);
		System.out.println(sb);
	}

	public static void dfs(int cnt, int cur) {
		sb.append(cur+" ");
		visit[cur] = true;
		for (int i=0; i<g[cur].size(); i++) {
			int tmp = (int) g[cur].get(i);
			if (!visit[tmp]) {
				dfs(cnt+1, tmp);
			}
		}
	}

	public static void bfs(int cur) {
		Queue<Integer> q = new ArrayDeque<>();
		q.offer(cur);
		visit[cur]=true;
		while (!q.isEmpty()) {
			int now = q.poll();
			sb.append(now+" ");
			for (int i=0; i<g[now].size(); i++) {
				int tmp = (int) g[now].get(i);
				if (!visit[tmp]) {
					q.offer(tmp);
					visit[tmp]=true;
				}
			}
		}
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

input = sys.stdin.readline

def dfs(g, visit, a):
    visit[a] = 1
    print(a, end=" ")
    for i in g[a]:
        if visit[i]==0:
            dfs(g, visit, i)


def bfs(g, visit, a):
    q = deque()
    q.append(a)
    visit[a] = 1
    while q:
        now = q.popleft()
        print(now, end=" ")
        for next in g[now]:
            if visit[next]==0:
                q.append(next)
                visit[next] = 1

n,m,v = map(int, input().split())
g = [[] for _ in range(n+1)]

for _ in range(m):
    a,b = map(int, input().split())
    g[a].append(b)
    g[b].append(a)
for i in range(1,n+1):
    g[i].sort()

visit = [0]*(n+1)
dfs(g, visit, v)
print()
visit = [0]*(n+1)
bfs(g, visit, v)